src/base/dimension.rs

```diff
@@ -343,3 +343,41 @@ from_to_typenum!(
     U111, U112, U113, U114, U115, U116, U117, U118, U119, U120, U121, U122, U123, U124, U125, U126,
     U127
 );
+
+pub trait DimUnify<D> {
+    type Output;
+
+    fn unify(self, other: D) -> Option<Self::Output>;
+}
```

Comment:
Yeah, this is probably a better idea. I think it should be possible to do this with a default method to avoid a breaking change. With the same technique I think it should also be doable with a freestanding  `unify` function. I don't know what would be best though.